src/libcore/array.rs

```diff
@@ -264,6 +267,269 @@ macro_rules! array_impls {
     }
 }
 
+#[cfg(not(bootstrap))]
+mod impls_using_const_generics {
+    use super::*;
+
+    #[stable(feature = "rust1", since = "1.0.0")]
+    impl<T, const N: usize> AsRef<[T]> for [T; N]
+        where [(); N]: AllowedArrayLength
+    {
+        #[inline]
+        fn as_ref(&self) -> &[T] {
+            &self[..]
+        }
+    }
+
+    #[stable(feature = "rust1", since = "1.0.0")]
+    impl<T, const N: usize> AsMut<[T]> for [T; N]
+        where [(); N]: AllowedArrayLength
```

Comment:
I’d say it’s worth running rustfmt on all this new code to have it be as consistent as possible to start